ch360/fileredactor.go

```diff
@@ -0,0 +1,55 @@
+package ch360
+
+import (
+	"context"
+	"github.com/CloudHub360/ch360.go/ch360/request"
+	"io"
+)
+
+type FileRedactor struct {
+	docCreator   DocumentCreator
+	docExtractor DocumentExtractor
+	docRedactor  DocumentRedactor
+	docDeleter   DocumentDeleter
+}
+
+func NewFileRedactor(creator DocumentCreator, extractor DocumentExtractor,
+	redactor DocumentRedactor,
+	deleter DocumentDeleter) *FileRedactor {
+	return &FileRedactor{
+		docCreator:   creator,
+		docExtractor: extractor,
+		docRedactor:  redactor,
+		docDeleter:   deleter,
+	}
+}
+
+func (f *FileRedactor) Redact(ctx context.Context, fileContents io.Reader,
+	extractorName string) (io.ReadCloser, error) {
+
+	// Use a different context here so we don't cancel this req on ctrl-c. We need
+	// the docId result to perform cleanup
+	document, err := f.docCreator.Create(context.Background(), fileContents)
+	if err != nil {
+		return nil, err
+	}
+
+	defer func() {
+		if document.Id != "" {
```

Comment:
💪 Again, I suspect this is duplicated many times. Can it be pushed down into some kind of abstraction?